ZipArchive.cs

```diff
@@ -840,12 +840,17 @@ public void Close ()
 			if (archive == IntPtr.Zero)
 				return;
 
-			Native.zip_close (archive);
-			foreach (var s in sources) {
-				s.Dispose ();
+			try {
+				if (Native.zip_close (archive) < 0)
+					throw GetErrorException ();
+			}
+			finally {
```

Comment:
If you could just put `finally` hanging behind the brace on the line above then LGTM :)

Reply:
fixed